src/mscorlib/src/System/Runtime/Intrinsics/X86/Sse41.cs

```diff
@@ -343,6 +361,28 @@ public static class Sse41
         /// </summary>
         public static Vector128<float> RoundCurrentDirection(Vector128<float> value) => RoundCurrentDirection(value);
 
+        /// <summary>
+        /// __m128 _mm_round_ss (__m128 a, int rounding)
+        /// _MM_FROUND_TO_NEAREST_INT |_MM_FROUND_NO_EXC
+        /// </summary>
+        public static Vector128<float> RoundToNearestIntegerScalar(Vector128<float> value) => RoundToNearestIntegerScalar(value);
```

Comment:
It would be more consistent to use a RoundingMode immediate parameter here (similarly to comparisons for example). Then it would be 4 functions (Round/RoundScalar * float/double) that map directly to four machine instructions (roundpd/ps/sd/ss) instead of 20. The fully named helper functions could be defined on top of these basic instructions somewhere else.

Reply:
@fiigii, thoughts? I was following the existing convention you had setup for the packed forms.

Reply:
AFAIR in discussions on Intrinsics API the consensus was to create direct mapping to processor instructions due to mutlitude of reasons. Therefore, I would avoid creating any APIs which do not map or omit any processor instructions. It means that if we have 3 argument scalar AVX or above instruction while having 2 argument SSE equivalent we should have both or here we should use immediate parameter for defining rounding mode.

Reply:
I believe that the current design (encodes rounding mode into intrinsic names) has better static semantics.
For example, RoundingMode immediate parameter requires 1) `const` parameter from language feature support to avoid non-literal values, 2) compile error reporting (or runtime exception) for invalid values from Roslyn/CoreCLR https://github.com/dotnet/corefx/issues/22940#issuecomment-320122766. 
Each round just has a few pre-defined modes, so I thought this is a good opportunity to provide intrinsic with more stable runtime behaviors and friendly development experience. Meanwhile, it does not lose any flexibility.

Reply:
> if this simplifies the implementation.

No, the current runtime always expands all the APIs as intrinsics.